jpo-ode-svcs/src/main/java/us/dot/its/jpo/ode/coder/stream/LogFileToAsn1CodecPublisher.java

```diff
@@ -156,76 +154,67 @@ public boolean isSpatRecord() {
-						msgMetadata.addEncoding(msgEncoding);
-					}
+				// Determine the message type (MAP or TIM are the current other options)
+				String messageType = determineMessageType(msgPayload);
+				if (messageType == "MAP") {
+					publisher.publish(JsonUtils.toJson(odeData, false),
+						publisher.getOdeProperties().getKafkaTopicOdeRawEncodedMAPJson());
+				} else if (messageType == "TIM") {
+					publisher.publish(JsonUtils.toJson(odeData, false),
+						publisher.getOdeProperties().getKafkaTopicOdeRawEncodedTIMJson());
 				}
-
-				Asn1Encoding unsecuredDataEncoding = new Asn1Encoding("unsecuredData", "MessageFrame",
-						EncodingRule.UPER);
-				msgMetadata.addEncoding(unsecuredDataEncoding);
-
-				publisher.publish(xmlUtils.toXml(odeData),
-						publisher.getOdeProperties().getKafkaTopicAsn1DecoderInput());
 			}
+
 			serialId.increment();
 		}
 	}
 
-	
 	/**
-		* Checks the header of the OdeMsgPayload and determines the encoding rule to be used in the Asn1DecoderInput XML.
-		* The payload is checked for various message start flags. It will add the encoding rule to the Asn1DecoderInput XML
-		* to tell the ASN1 codec to extract data from the header into the output message.
+		* Determines the message type based off the most likely start flag
 		* 
-		* @param payload The OdeMsgPayload to check the header of.
-		* @return The encoding rule to be used in the Asn1DecoderInput XML.
+		* @param payload The OdeMsgPayload to check the content of.
 		*/
-	public String checkHeader(OdeMsgPayload payload) {
-		JSONObject payloadJson;
-		String header = null;
+	public String determineMessageType(OdeMsgPayload payload) {
+		String messageType = "";
 		try {
-			payloadJson = JsonUtils.toJSONObject(payload.getData().toJson());
-			String hexPacket = payloadJson.getString("bytes");
-
-			for (String key : msgStartFlags.keySet()) {
-				String startFlag = msgStartFlags.get(key);
-				int startIndex = hexPacket.toLowerCase().indexOf(startFlag);
-				logger.debug("Start index for " + key + "(" + startFlag + ")" + " is: " + startIndex);
-				if (startIndex <= 20 && startIndex != 0 && startIndex != -1) {
-					logger.debug("Message has supported Ieee1609Dot2Data header, adding encoding rule to Asn1DecoderInput XML");
-					header = "Ieee1609Dot2Data";
-					break;
+			JSONObject payloadJson = JsonUtils.toJSONObject(payload.getData().toJson());
+			String hexString = payloadJson.getString("bytes").toLowerCase();
+
+			HashMap<String, Integer> flagIndexes = new HashMap<String, Integer>();
+			flagIndexes.put("MAP", hexString.indexOf(msgStartFlags.get("MAP")));
+			flagIndexes.put("TIM", hexString.indexOf(msgStartFlags.get("TIM")));
+
+			int lowestIndex = Integer.MAX_VALUE;
+			for (String key : flagIndexes.keySet()) {
+				if (flagIndexes.get(key) == -1) {
+					logger.debug("This message is not of type " + key);
+					continue;
+				}
+				if (flagIndexes.get(key) < lowestIndex) {
```

Comment:
Will this ever not be true? Also, should we be concerned about finding the start string in the payload?

Reply:
It is as good as the previous check that was being performed and it is the best shot we got if I am entirely honest.

Reply:
A better way to get this to work might be to spin up an instance of wireshark and stream the kafka traffic through it and parse its output in real time. We would need to figure out how to get wireshark to properly read the ASN1 which I haven't succeeded at doing. We could then use it to figure out where the headers begin and end. The most we are going to do is go off of guess work.

Reply:
What do you think of looking for all payload start strings and then if multiple are found then choose the one that occurs first? This should limit the risk of picking the wrong message type, but would sacrifice a bit of unnecessary compute most of the time.

Reply:
okay nvm I wasn't looking at your logic where you would update the lowest index. Looks good!